macOS build failure: hvf_darwin.go:174:3: could not determine what C.HV_MEMORY_NON_SECURE r

## macOS build failure

```
# github.com/dalinkstone/tent/internal/hypervisor/hvf
internal/hypervisor/hvf/hvf_darwin.go:174:3: could not determine what C.HV_MEMORY_NON_SECURE refers to
internal/hypervisor/hvf/hvf_darwin.go:160:3: could not determine what C.MAP_ANON refers to
internal/hypervisor/hvf/hvf_darwin.go:164:18: could not determine what C.MAP_FAILED refers to
internal/hypervisor/hvf/hvf_darwin.go:160:14: could not determine what C.MAP_PRIVATE refers to
internal/hypervisor/hvf/hvf_darwin.go:159:3: could not determine what C.PROT_READ refers to
internal/hypervisor/hvf/hvf_darwin.go:159:15: could not determine what C.PROT_WRITE refers to
internal/hypervisor/hvf/hvf_darwin.go:171:3: could not determine what C.hv_gpa_t refers to
internal/hypervisor/hvf/hvf_darwin.go:172:3: could not determine what C.hv_ra_t refers to
internal/hypervisor/hvf/hvf_darwin.go:62:11: could not determine what C.hv_vm_t refers to
internal/hypervisor/hvf/hvf_darwin.go:156:15: could not determine what C.mmap refers to
internal/hypervisor/hvf/hvf_darwin.go:177:3: could not determine what C.munmap refers to
cgo: 
cc errors for preamble:
internal/hypervisor/hvf/hvf_darwin.go:22:14: error: use of undeclared identifier 'HV_NO_RESOURCE'; did you mean 'HV_NO_RESOURCES'?
   22 |         case HV_NO_RESOURCE:
      |              ^~~~~~~~~~~~~~
      |              HV_NO_RESOURCES
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/arm64/hv/hv_kern_types.h:56:2: note: 'HV_NO_RESOURCES' declared here
   56 |         HV_NO_RESOURCES        = err_common_hypervisor | 0x05, /* (0xfae94005) */
      |         ^
internal/hypervisor/hvf/hvf_darwin.go:24:14: error: use of undeclared identifier 'HV_NO_SPACE'; did you mean 'HV_NO_DEVICE'?
   24 |         case HV_NO_SPACE:
      |              ^~~~~~~~~~~
      |              HV_NO_DEVICE
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/arm64/hv/hv_kern_types.h:57:2: note: 'HV_NO_DEVICE' declared here
   57 |         HV_NO_DEVICE           = err_common_hypervisor | 0x06, /* (0xfae94006) */
      |         ^
internal/hypervisor/hvf/hvf_darwin.go:28:14: error: use of undeclared identifier 'HV_IN_USE'
   28 |         case HV_IN_USE:
      |              ^
internal/hypervisor/hvf/hvf_darwin.go:30:14: error: use of undeclared identifier 'HV_NO_PRIVILEGE'
   30 |         case HV_NO_PRIVILEGE:
      |              ^
internal/hypervisor/hvf/hvf_darwin.go:32:14: error: use of undeclared identifier 'HV_IO_ERROR'; did you mean 'HV_ERROR'?
   32 |         case HV_IO_ERROR:
      |              ^~~~~~~~~~~
      |              HV_ERROR
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/arm64/hv/hv_kern_types.h:52:2: note: 'HV_ERROR' declared here
   52 |         HV_ERROR               = err_common_hypervisor | 0x01, /* (0xfae94001) */
      |         ^
internal/hypervisor/hvf/hvf_darwin.go:34:14: error: use of undeclared identifier 'HV_VM_CONFIG_ERROR'
   34 |         case HV_VM_CONFIG_ERROR:
      |              ^
6 errors generated.
```

## Platform
Darwin arm64, Go go1.26.1

## How to reproduce
```bash
cd project && go build -o tent ./cmd/tent
```

## Priority
This blocks all macOS testing. Fix the compilation errors first, then move on to features.

--- a/project/internal/hypervisor/hvf/hvf_darwin.go
+++ b/project/internal/hypervisor/hvf/hvf_darwin.go
@@ -166,12 +166,13 @@ func (v *VM) Start() error {
 	}
 
 	// Map memory to VM
+	// Note: hv_vm_map signature is (uva, gpa, size, flags) - no vmRef parameter
+	// The VM is implicitly associated with the current task after hv_vm_create
 	ret = C.hv_vm_map(
-		v.vmRef,
-		C.hv_gpa_t(0),
-		C.hv_ra_t(unsafe.Pointer(memoryPtr)),
-		memorySize,
-		C.HV_MEMORY_NON_SECURE,
+		C.hv_uvaddr_t(unsafe.Pointer(memoryPtr)),
+		C.hv_gpaddr_t(0),
+		C.size_t(memorySize),
+		C.HV_MEMORY_READ|C.HV_MEMORY_WRITE|C.HV_MEMORY_EXEC,
 	)
 	if ret != C.HV_SUCCESS {
 		C.munmap(memoryPtr, C.size_t(memorySize))

